### DISCUSSION TITLE
ZZH problems

### DISCUSSION THREAD

**Discussion by 2N3819** [Koenkk/zigbee2mqtt#7381] [Devices]
Hi
I have recently upgraded from a CC2531 co-ordinator to a ZZH co-ordinator.
I started with a clean install of zigbee2mqtt at the latest stable version.

info  2021-05-05 20:52:49: Starting Zigbee2MQTT version 1.18.3 (commit #f2e39af1)
info  2021-05-05 20:52:49: Starting zigbee-herdsman (0.13.92)
info  2021-05-05 20:52:52: zigbee-herdsman started
info  2021-05-05 20:52:52: Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20210120,"transportrev":2},"type":"zStack3x0"}

After approx 24 hours of running zigbee2mqtt logs the following errors:

debug 2021-05-06 23:10:00: Received MQTT message on 'zigbee2mqtt/EweLink FR TV/set' with data '{"state": "OFF"}'
debug 2021-05-06 23:10:00: Publishing 'set' 'state' to 'EweLink FR TV'
error 2021-05-06 23:10:06: Publish 'set' 'state' to 'EweLink FR TV' failed: 'Error: Command 0x00124b00226ab03a/1 genOnOff.off({}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (SRSP - AF - dataRequest after 6000ms)'
debug 2021-05-06 23:10:06: Error: Command 0x00124b00226ab03a/1 genOnOff.off({}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (SRSP - AF - dataRequest after 6000ms)
    at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:37:35)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)
info  2021-05-06 23:10:06: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'state' to 'EweLink FR TV' failed: 'Error: Command 0x00124b00226ab03a/1 genOnOff.off({}, {\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (SRSP - AF - dataRequest after 6000ms)'","meta":{"friendly_name":"EweLink FR TV"},"type":"zigbee_publish_error"}' 

Other devices are also affected:
info  2021-05-07 00:00:12: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'state' to 'FROutsideLights' failed: 'Error: Command 1 genOnOff.off({}) failed (SRSP - AF - dataRequestExt after 6000ms)'","meta":{"friendly_name":"FROutsideLights"},"type":"zigbee_publish_error"}'
error 2021-05-07 00:00:18: Publish 'set' 'state' to 'SP222 landing' failed: 'Error: Command 0x14b457fffe4202d5/1 genOnOff.off({}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (SRSP - AF - dataRequest after 6000ms)'
debug 2021-05-07 00:00:18: Error: Command 0x14b457fffe4202d5/1 genOnOff.off({}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (SRSP - AF - dataRequest after 6000ms)
    at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:37:35)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)
info  2021-05-07 00:00:18: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'state' to 'SP222 landing' failed: 'Error: Command 0x14b457fffe4202d5/1 genOnOff.off({}, {\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (SRSP - AF - dataRequest after 6000ms)'","meta":{"friendly_name":"SP222 landing"},"type":"zigbee_publish_error"}'
debug 2021-05-07 00:01:00: Received MQTT message on 'zigbee2mqtt/Heating/set' with data '{"state": "OFF"}'
debug 2021-05-07 00:01:00: Publishing 'set' 'state' to 'Heating'
debug 2021-05-07 00:01:01: Received MQTT message on 'zigbee2mqtt/Hot Water/set' with data '{"state": "ON"}'
debug 2021-05-07 00:01:01: Publishing 'set' 'state' to 'Hot Water'
error 2021-05-07 00:01:06: Publish 'set' 'state' to 'Heating' failed: 'Error: Command 0x00124b001e72891c/1 genOnOff.off({}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (SRSP - AF - dataRequest after 6000ms)'
debug 2021-05-07 00:01:06: Error: Command 0x00124b001e72891c/1 genOnOff.off({}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (SRSP - AF - dataRequest after 6000ms)
    at Timeout._onTimeout (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:37:35)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)

Zibgee2mqtt keeps running but appears slow to switch devices on and off.

My configuration.yaml is:
homeassistant: false
permit_join: true
frontend:
  port: 8080
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://localhost'
serial:
  port: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
advanced:
  rtscts: false
  pan_id: 6755
  log_level: debug
devices:
  '0x14b457fffe4b1117':
    friendly_name: SP222 Hall
  '0x000d6ffffe5cbcba':
    friendly_name: sp222 no1
  '0x14b457fffe4afb79':
    friendly_name: SP222 kitchen
  '0x14b457fffe4202d5':
    friendly_name: SP222 landing
  '0x00124b001e72899e':
    friendly_name: Hot Water
  '0x00124b001e72891c':
    friendly_name: Heating
  '0x847127fffe24e0eb':
    friendly_name: Front Floodlight
  '0x847127fffe24dfac':
    friendly_name: Front PIR
  '0x04cf8cdf3c7cc24b':
    friendly_name: Light Sensor
  '0x7cb03eaa0a04bc9d':
    friendly_name: F Room out R
  '0x7cb03eaa0a034813':
    friendly_name: F Room out L
  '0x00158d000429f6ca':
    friendly_name: Hall Temp
  '0x00124b00226ab03a':
    friendly_name: EweLink FR TV
  '0x00124b001fc9388f':
    friendly_name: Cat Bell
  '0x00124b00226865e9':
    friendly_name: Zigbee temp 2
groups:
  '1':
    friendly_name: FROutsideLights
    devices:
      - 0x7cb03eaa0a034813/3
      - 0x7cb03eaa0a04bc9d/3

Any advice would be appreciated.

**Comment by 2N3819** [comment]
The problem appears to be the ZZH which is a "diva" on a Raspberry Pi.

Not only is not compatible with any other USB clients being attached to the Pi (I am guessing because it does not rtccts, this has to be set to false), the ZZH appears to be unstable when the Pi's bus is in the normal configuration of supporting both USB1.1 and 2. 

I found that the ZZH was dropping off the USB bus.
From dmesg:
ch341-uart ttyUSB0: usb_serial_generic_read_bulk_callback - urb stopped: -32

From https://github.com/raspberrypi/linux/issues/2406 
I configured the Pi to USB 1.1 only mode Set dwc_otg.speed=1 in /boot/cmdline.txt 

The ZZH now appears to be stable.
However, the ZZH still needs a manual reset (short press of the button switch on the ZZH) when the ZZH has been removed from the USB bus or the Pi is rebooted.

**Comment by Matthy1810** [comment]
Did you try to connect the stick via an extension cable to the rpi? I think this can reduce interference.

**Comment by 2N3819** [comment]
Yes I always use a 0.5 metre USB extension cable.
The ZZH is now stable after a reset at start up.
The Pis bluetooth and WiFi are disabled in  /boot/cmdline.txt
I will try a longer USB extenstion cable.
